src/teleop/jetson/jetson_teleop.py

```diff
@@ -97,37 +100,108 @@ def __init__(self):
+        :param quad_control: Bool for whether or not we want the axis to follow an x^2 curve instead of a linear one
+        Velocities are sent in range [-1,1]
+        :return: Returns an output velocity value for the given joint using the given axis_name
+        """
+        return (
+            quadratic(deadzone(axes_array[self.xbox_mappings[axis_name]], deadzone_threshold))
+            if quad_control
+            else deadzone(axes_array[self.xbox_mappings[axis_name]], deadzone_threshold)
+        )
+
+    def filter_xbox_button(self, button_array: "list[int]", pos_button: str, neg_button: str) -> float:
+        """
+        Applies various filtering functions to an axis for controlling the arm
+        :param button_array: Button array from sensor_msgs/Joy, each value is an int 0 or 1
+        :param pos_button: String representing the positive button for controlling a joint
+        :param neg_button: String representing the negtaive button for controlling a joint
+        :return: Return -1, 0, or 1 depending on what buttons are being pressed
+        """
+        return button_array[self.xbox_mappings[pos_button]] - button_array[self.xbox_mappings[neg_button]]
+
+    def ra_control_callback(self, msg: Joy) -> NoReturn:
+        """
+        Converts a Joy message with the Xbox inputs
+        to a JointState to control the RA Arm in open loop
+        :param msg: Has axis and buttons array for Xbox controller
+        Velocities are sent in range [-1,1]
+        :return:
+        """
+
+        # Filter for xbox triggers, they are typically [-1,1]
         raw_left_trigger = msg.axes[self.xbox_mappings["left_trigger"]]
         left_trigger = raw_left_trigger if raw_left_trigger > 0 else 0
         raw_right_trigger = msg.axes[self.xbox_mappings["right_trigger"]]
         right_trigger = raw_right_trigger if raw_right_trigger > 0 else 0
+
         self.ra_cmd.velocity = [
-            self.ra_config["joint_a"]["multiplier"]
-            * quadratic(deadzone(msg.axes[self.xbox_mappings["left_js_x"]], 0.15)),
-            self.ra_config["joint_b"]["multiplier"]
-            * quadratic(-deadzone(msg.axes[self.xbox_mappings["left_js_y"]], 0.15)),
-            self.ra_config["joint_c"]["multiplier"]
-            * quadratic(-deadzone(msg.axes[self.xbox_mappings["right_js_y"]], 0.15)),
-            self.ra_config["joint_d"]["multiplier"]
-            * quadratic(deadzone(msg.axes[self.xbox_mappings["right_js_x"]], 0.15)),
+            self.ra_config["joint_a"]["multiplier"] * self.filter_xbox_axis(msg.axes, "left_js_x", 0.15, True),
+            self.ra_config["joint_b"]["multiplier"] * self.filter_xbox_axis(msg.axes, "left_js_y", 0.15, True),
+            self.ra_config["joint_c"]["multiplier"] * self.filter_xbox_axis(msg.axes, "right_js_y", 0.15, True),
+            self.ra_config["joint_d"]["multiplier"] * self.filter_xbox_axis(msg.axes, "right_js_x", 0.15, True),
```

Comment:
Feels like there's gotta be a decent way to do this with less duplication. Not sure the best way to do it right now

Reply:
Yeah I felt this way too, but trying to abstract more of this into functions ended up just being a function with a ton of params. Could maybe use a list interpretation to clean it up a bit?